x-pack/plugins/observability_solution/inventory/public/components/grouped_inventory/index.tsx

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { EuiSpacer, EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import React from 'react';
+import useEffectOnce from 'react-use/lib/useEffectOnce';
+import { ENTITY_TYPE } from '@kbn/observability-shared-plugin/common';
+import { useInventorySearchBarContext } from '../../context/inventory_search_bar_context_provider';
+import { useInventoryAbortableAsync } from '../../hooks/use_inventory_abortable_async';
+import { useInventoryRouter } from '../../hooks/use_inventory_router';
+import { useKibana } from '../../hooks/use_kibana';
+import { GroupSelector } from './group_selector';
+import { InventoryGroupAccordion } from './inventory_group_accordion';
+import { groupCountCss } from './styles';
+import { useInventoryParams } from '../../hooks/use_inventory_params';
+
+export interface GroupedInventoryPageProps {
+  setGroupSelected: (selected: string) => void;
+}
+
+export function GroupedInventoryView() {
+  const { searchBarContentSubject$ } = useInventorySearchBarContext();
+  const {
+    services: { inventoryAPIClient },
+  } = useKibana();
+
+  const inventoryRoute = useInventoryRouter();
+  const { query } = useInventoryParams('/');
+  const { kuery, entityTypes } = query;
+
+  const { value = { groupBy: ENTITY_TYPE, groups: [] }, refresh } = useInventoryAbortableAsync(
+    ({ signal }) => {
+      return inventoryAPIClient.fetch('GET /internal/inventory/entities/group_by/{field}', {
+        params: {
+          path: {
+            field: ENTITY_TYPE,
+          },
+          query: {
+            kuery,
+            entityTypes: entityTypes?.length ? JSON.stringify(entityTypes) : undefined,
+          },
+        },
+        signal,
+      });
+    },
+    [entityTypes, inventoryAPIClient, kuery]
+  );
+
+  const totalEntities = value.groups.reduce((acc, group) => acc + group.count, 0);
+
+  useEffectOnce(() => {
+    const searchBarContentSubscription = searchBarContentSubject$.subscribe(
+      ({ refresh: isRefresh, ...queryParams }) => {
+        if (isRefresh) {
+          refresh();
+        } else {
+          inventoryRoute.push('/', {
+            path: {},
+            query: { ...query, ...queryParams },
+          });
+        }
+      }
+    );
+    return () => {
+      searchBarContentSubscription.unsubscribe();
+    };
+  });
```

Comment:
You can have this defined in a single place (`public/pages/inventory_page/index.tsx`).

Reply:
I move these to the inventory_page, but I wasn't quite sure how to deal with the `refresh()` method dependency. should I just remove it?

Reply:
I was giving a though about this too. And maybe you could export a new observable that will only be invoked when refresh happens? Then this component and the unified page component would only subscribe to this event and call refresh.

Reply:
I've refactored the search bar context to also have a refresh subject, so now downstream API calls can subscribe to that subject instead in order to refresh. This keeps the SearchBarSubscription defined once at the top level, but it now has to be a normal `useEffect` instead of just `useEffectOnce` in order to track changes to `query`. Will this be okay or will React's redo everything on re-render approach to components have adverse effects here?